.github/workflows/asan.yml

```
@@ -0,0 +1,70 @@
+name: Address Sanitizer analysis
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '0 1 * * *'
+
+jobs:
+
+  asan-test:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Sync eProsima/Fast-DDS repository
+        uses: actions/checkout@v3
+        with:
+          path: src/Fast-DDS
+
+      - name: Install apt packages
+        uses: ./src/Fast-DDS/.github/actions/install-apt-packages
+
+      - name: Install GTest
+        uses: ./src/Fast-DDS/.github/actions/install-gtest-linux
+
+      - name: Install Python packages
+        uses: ./src/Fast-DDS/.github/actions/install-python-packages
+
+      - name: Update colcon mixin
+        run: |
+          colcon mixin add default \
+            https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
+          colcon mixin update default
+        continue-on-error: true
+
+      - name: Fetch Fast DDS dependencies
+        uses: ./src/Fast-DDS/.github/actions/fetch-fastdds-repos
+
+      - name: Build workspace
+        run: |
+          cat src/Fast-DDS/.github/workflows/asan_colcon.meta
+          colcon build \
+            --event-handlers=console_direct+ \
+            --metas src/Fast-DDS/.github/workflows/asan_colcon.meta
+
+      - name: Run tests Fast DDS
+        run: |
+          source install/setup.bash && \
+          colcon test \
+            --packages-select fastrtps \
+            --event-handlers=console_direct+ \
+            --return-code-on-test-failure \
+            --ctest-args --timeout 60
```

```suggestion
            --return-code-on-test-failure \
            --ctest-args --label-exclude xfail --timeout 60
```
Probably xfail tests (and maybe some others) must be skipped